gnat2goto/driver/tree_walk.adb

```diff
@@ -1007,109 +1008,254 @@ package body Tree_Walk is
    -- Do_Array_Assignment --
    -------------------------
 
-   --  TODO: multi-dimensional arrays
-   function Do_Array_Assignment (LHS_Expr : Irep; RHS_Expr : Irep; N : Node_Id)
-                                return Irep
+   --  The following function builds a generalised array assignment
+   --  dest := src_1 & src_2 & .. & src_n
+   --  for $n>0$, src_i may overlap with dest
```

Comment:
I interpret this statement as being trivially true (n must be greater than 0 or you have no source and no assignment)
even in case of the identity assignment, your source would exist. 
Perhaps in this case less is more:
"src_i may overlap with dest" or "src_i may be part of dest"

Reply:
Yes, the $n>0$ relates to the line above, which I guess is confusing. I'll rewrite it.